service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseDataExtractionService.java

```diff
@@ -195,14 +196,16 @@ private String getFirstApplicantName(CaseData caseData) {
             .orElse("");
     }
 
-    private Map<String, List<Map<String, String>>> getGroupedDirections(CaseData caseData) throws IOException {
+    private ImmutableMap<String, List<Map<String, String>>> getGroupedDirections(CaseData caseData)
+        throws IOException {
```

Comment:
why?

Reply:
should be Map<DirectionAssignee, List...> here no?

Reply:
the thing we return from this is getting added to case data with data.putAll(), so don't we need our key of the map to be a string? if we use Map<DIrectionAssignee, ... then we need to stream it so we can getKey().getValue() inside putAll(). can do either approach

Reply:
Could we possibly annotate the `getValue` method in the enum with `@JsonValue`?

Reply:
I think this would then serialise the enum using the result of that method when it is being converted in json

Reply:
this is the method for getting the data for the docmosis template - the docmosis service expects a string, so if we change to DirectionAssignee then docmosis cannot render it. there's no JSON conversion so we can't use JsonValue. I think we have to keep it as is